name: CITest

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  validate-inspire:
    runs-on: ubuntu-latest
    services:
      inspire:
        image: docker.pkg.github.com/inspire-eu-validation/community/inspire-validator:2020.3
        credentials:
          username: koalapdok
          password: ${{ secrets.PDOK_KOALA_PERSONAL_ACCESS_TOKEN }}
        ports: 
          - 8080:8080
    steps:
      - uses: actions/checkout@v2
      - name: Check status of inspire validator
        # Dit zal wel niet werken omdat de service er even over doet om op te starten
        run: |
          echo "start"
          getstate() {
            curl --fail -X GET --header 'Accept: application/json' 'localhost:8080/validator/v2/status'
            return $?
          }

          max_retry=6
          counter=0

          until getstate
          do
             sleep 10
             [[ counter -eq $max_retry ]] && echo "Could not connect to inspire validation" && exit 1
             #echo "Trying again. Try #$counter"#use to debug
             ((counter++))
          done

          echo "Inspire service is up"

